mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -136,14 +136,20 @@ def parse_json_input(json_input, orient="split", schema: Schema = None):
         )
 
 
-def parse_csv_input(csv_input):
+def parse_csv_input(csv_input, schema: Schema = None):
     """
     :param csv_input: A CSV-formatted string representation of a Pandas DataFrame, or a stream
                       containing such a string representation.
+    :param schema: Optional schema specification to be used during parsing.
     """
 
     try:
-        return pd.read_csv(csv_input)
+        csv_input = StringIO(csv_input)
```

Comment:
Let's move this line back to  https://github.com/mlflow/mlflow/pull/5531/files#diff-e607d3e6aefa036b1bfe9416fd1aa8b0934120f458cf263ddebab224f832e4a5L264 since the input to parse_csv can also be a path to a csv file. We need to fix our documentation for this.